### PR TITLE
Link CFITSIO library with cmake

### PR DESCRIPTION
**Pull request by aderylo**
## Category:
Please pick one from below:
**New feature** (*non-breaking change which adds functionality*)

## Description:
- Adds linking of cfitsio in cmake, so it can be used in FITS file reader/parser.   
- Context: MIMUW ZPP project: GPU accelerated FITS RICE decoding

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ X] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
Would it make sense to write a simple tests, for example one that tries to create dummy FITS file? To see if the integration works.

**Comment by klecki** [comment]
Also, the license needs to be added here: https://github.com/NVIDIA/DALI/blob/main/Acknowledgements.txt

**Comment by szalpal** [comment]
Please adjust the title of the PR to be more descriptive.

Also, should we add an acknowledgement, according to the license: https://github.com/healpy/cfitsio/blob/316e95008492b597b3cbcf84168df22996fe2b6f/License.txt#L6. I believe `Acknowledgement.txt` will be a good place for it.

File: CMakeLists.txt
Comment (mzient) [inline CMakeLists.txt:78]: If we don't ship the library (either in `third_party` or `DALI_deps`), then we should enable it by default.
Alternatively, we can just try to use it and disable (instead of giving a hard error) when it's not found.

Comment (aderylo) [reply CMakeLists.txt:78]: I think that the plan is to ship it with DALI_deps (https://github.com/NVIDIA/DALI_deps/pull/68) , although  I don't know if enabling it by default is a way to go since it will be only needed for FITS reader. What do you recon?

**Comment by aderylo** [comment]
> Would it make sense to write a simple tests, for example one that tries to create dummy FITS file? To see if the integration works.

Sure, I've added some simple tests of basic cfitsio rutines in /dali/test/cfitsio_test.cc

File: CMakeLists.txt
Comment (mzient) [reply CMakeLists.txt:78]: OK, then add the commit hash (from your GH repo) to this PR - then we can run it in CI and test that it compiles and works. After that we can merge the changes to DALI extra and specify the proper commit hash in this PR. Then it will be mergeable.

**Comment by klecki** [comment]
Please fix the linter errors:
```
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:18:  Include the directory when naming .h files  [build/include] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:24:  Do not indent within a namespace  [runtime/indentation_namespace] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:25:  protected: should be indented +1 space inside class CfitsioTest  [whitespace/indent] [3]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:25:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:26:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:27:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:28:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:29:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:30:  Missing space before {  [whitespace/braces] [5]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:31:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:32:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:36:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:38:  Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:39:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:40:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:46:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:47:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:51:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:55:  Missing space before {  [whitespace/braces] [5]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:56:  Almost always, snprintf is better than strcpy  [runtime/printf] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:59:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:65:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:67:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
#14 56.31 /opt/dali/dali/test/cfitsio_test.cc:67:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
#14 56.31 
```
You can run linter on your own by invoking `make lint` in the build directory.

File: dali/test/cfitsio_test.cc
Comment (klecki) [inline dali/test/cfitsio_test.cc:60]:
```suggestion

  ffflnm(fptr, filename, &status);
  EXPECT_EQ(status, 0);
  ffflmd(fptr, &filemode, &status);
  EXPECT_EQ(status, 0);
```

**Comment by klecki** [comment]
Hmm, I think we also need to adjust Conda build and dockerfiles for completeness. I will take a look into that a bit later and schedule another build.

**Review by klecki — APPROVED** [review]

**Comment by klecki** [comment]
!build

**Review by mzient — APPROVED** [review]

**Pull request merged**
